src/main/java/uk/gov/justice/probation/courtcaseservice/service/UserAgnosticOffenderService.java

```diff
@@ -40,20 +41,27 @@ public Mono<ProbationStatusDetail> getProbationStatusWithoutRestrictions(String
 
     public Optional<OffenderEntity> updateOffenderProbationStatus(String crn) {
         return offenderRepository.findByCrn(crn)
-                .map(offenderEntity -> {
-                    ProbationStatusDetail probationStatusDetailFromCommunityApi = getProbationStatusWithoutRestrictions(crn).block();
-                    if (probationStatusDetailFromCommunityApi != null && !Objects.equals(probationStatusDetailFromCommunityApi, offenderEntity.getProbationStatusDetail())) {
-                        updateProbationStatusDetails(probationStatusDetailFromCommunityApi, offenderEntity);
-                        Optional.of(offenderRepository.save(offenderEntity))
-                                .map(updatedOffender -> {
-                                    telemetryService.trackOffenderProbationStatusUpdateEvent(updatedOffender);
-                                    return updatedOffender;
-                                });
-                    } else {
-                        telemetryService.trackOffenderProbationStatusNotUpdateEvent(offenderEntity);
-                    }
-                    return offenderEntity;
-                });
+            .map(offenderEntity -> {
+                log.info("Fetching probation status for crn: ", crn);
+                ProbationStatusDetail probationStatusDetailFromCommunityApi = getProbationStatusWithoutRestrictions(crn).block();
+                if (isUpdateProbationStatus(offenderEntity, probationStatusDetailFromCommunityApi)) {
+                    updateProbationStatusDetails(probationStatusDetailFromCommunityApi, offenderEntity);
+                    Optional.of(offenderRepository.save(offenderEntity))
+                        .map(updatedOffender -> {
+                            telemetryService.trackOffenderProbationStatusUpdateEvent(updatedOffender);
+                            return updatedOffender;
+                        });
+                } else {
+                    telemetryService.trackOffenderProbationStatusNotUpdateEvent(offenderEntity);
+                }
+                return offenderEntity;
+            });
+    }
+
+    private static boolean isUpdateProbationStatus(OffenderEntity offenderEntity, ProbationStatusDetail probationStatusDetailFromCommunityApi) {
+        return probationStatusDetailFromCommunityApi != null &&
+            !Objects.equals(probationStatusDetailFromCommunityApi, offenderEntity.getProbationStatusDetail()) &&
+            !StringUtils.equalsIgnoreCase("NO_RECORD", probationStatusDetailFromCommunityApi.getStatus());
```

Comment:
Any reason why we do not want this status in our system?

Reply:
It is not a probation status, it is as good as 404. In CCS CRN null treated as no record